### PR TITLE
Moved string to ID conversion out of commands

### PR DESCRIPTION
**Pull request by antonovchtd**
Per @KharchenkoMaks suggestion, moved string to ID conversion out of actions (commands), into the HomeStep. The Validating commands now receive IDs as Core::TaskID, instead of as a string

### CODE REVIEW

File: src/ui/actions/GetTaskToShowLabelsAction.cpp [hunk 1]
```diff
@@ -4,22 +4,17 @@
...
-        id.set_value(std::stoi(arg_));
-        if (!model->IsPresent(id))
-            return {ActionResult::Status::ID_NOT_FOUND, id};
-    } catch (const std::invalid_argument &) {
-        return {ActionResult::Status::TAKES_ID, std::nullopt};
+    if (id_) {
+        if (!model->IsPresent(*id_))
+            return {ActionResult::Status::ID_NOT_FOUND, id_};
+    } else {
+        return {ActionResult::Status::TAKES_ID, id_};
```
Comment (KharchenkoMaks) [inline src/ui/actions/GetTaskToShowLabelsAction.cpp:15]: confusing result `TAKES_ID`, maybe better to use something like `ERROR_NO_ID` to show clearly, that ID wasn't provided to this action and it is an error.

File: src/ui/actions/GetTasksToShowAction.cpp [hunk 2]
```diff
@@ -4,24 +4,21 @@
...
+        return {ActionResult::Status::ID_NOT_FOUND, id_};
 
     std::vector<Core::TaskEntity> tasks;
-    if (arg_.empty())
+    if (!id_ && arg_.empty())
         tasks = model->getTasks();
-    else if (id)
-        tasks = model->getTaskWithSubtasks(*id);
+    else if (id_)
+        tasks = model->getTaskWithSubtasks(*id_);
```
Comment (KharchenkoMaks) [inline src/ui/actions/GetTasksToShowAction.cpp]: Why using if statement to define different behaviours? Consider creating different Actions, for showing by id, label or show all tasks. In this case, your GetTasksToShowActions changes it's behaviour depending on its argument (id or string - label).

**Review by KharchenkoMaks — CHANGES_REQUESTED** [review]

File: src/ui/actions/GetTasksToShowAction.cpp (hunk 2 above)
Comment (antonovchtd) [reply src/ui/actions/GetTasksToShowAction.cpp]: good point, fixed

File: src/ui/actions/GetTaskToShowLabelsAction.cpp (hunk 1 above)
Comment (antonovchtd) [reply src/ui/actions/GetTaskToShowLabelsAction.cpp:15]: will think about it in "reworking ActionResult" PR. (TAKES_ID short for "takes an ID as an argument")

**Review by IlliaPlaksa — APPROVED** [review]

**Review by AntonApraksin — APPROVED** [review]

**Review by vladimir-moiseiev — APPROVED** [review]

**Pull request merged**